pkg/resources/rbac.authorization.k8s.io/v1/role/validator_test.go

```diff
@@ -0,0 +1,128 @@
+package role
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	gvk = metav1.GroupVersionKind{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "Role"}
+	gvr = metav1.GroupVersionResource{Group: "rbac.authorization.k8s.io", Version: "v1", Resource: "roles"}
+)
+
+func TestAdmit(t *testing.T) {
+	t.Parallel()
+
+	defaultRole := &v1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+	}
+	roleWithOwnerLabel := &v1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+			Labels: map[string]string{
+				grOwnerLabel: "gr-owner",
+			},
+		},
+	}
+
+	type args struct {
+		oldRole *v1.Role
+		newRole *v1.Role
+		op      admissionv1.Operation
+	}
+	type test struct {
+		name    string
+		args    args
+		allowed bool
+		wantErr bool
+	}
+	tests := []test{
```

Comment:
I'd also recommend the following test cases:
- Old Role labels nil, new role has something -> allowed
- Old Role labels nil, new role labels nil (I know the selector prevents this, but the `admit` function should still not panic) -> allowed
- Old Role labels empty (`map[string]string{}`), new role has something -> allowed
- Old role has something, new role has labels empty (`map[string]string{}`) -> denied
- Old role labels empty (`map[string]string{}`), new role labels empty (`map[string]string{}`) -> allowed.